widget/src/button.rs

```diff
@@ -173,11 +182,66 @@ where
         self.class = class.into();
         self
     }
+
+    /// Sets the [`AnimationDuration`] forward duration (in milliseconds) of the [`Button`].
+    pub fn animation_forward_duration(
+        mut self,
+        animation_duration_ms: f32,
+    ) -> Self {
+        self.animation_forward_duration.set(animation_duration_ms);
+        self
+    }
+
+    /// Sets the [`AnimationDuration`] backward duration (in milliseconds) of the [`Button`].
+    pub fn animation_backward_duration(
+        mut self,
+        animation_duration_ms: f32,
+    ) -> Self {
+        self.animation_backward_duration.set(animation_duration_ms);
+        self
+    }
+
+    /// Enables or disables [`AnimationDuration`] transition on [`Pressed`] and [`Hover`] states.
+    pub fn set_animations_enabled(mut self, enable: bool) -> Self {
+        if enable {
+            self.animation_forward_duration.enable();
+            self.animation_backward_duration.enable();
+        } else {
+            self.animation_forward_duration.disable();
+            self.animation_backward_duration.disable();
+        }
+        self
+    }
```

Comment:
What does this code do that isn't available by default in lilt?
I'm wondering if this is something that could be addressed there 🙂

Reply:
These are just some wrapper functions meant to abstract away the animations internals for the user, and they're also used in the `State` creation (see references to `AnimationDuration::get()`).